src/bin/pg_autoctl/monitor.c

```diff
@@ -661,6 +661,12 @@ printCurrentState(void *ctx, PGresult *result)
 	/* prepare a nice dynamic string of '-' as a header separator */
 	nameSeparatorHeader = (char *) malloc((maxNodeNameSize+1) * sizeof(char));
 
+	if (nameSeparatorHeader == NULL) {
+		log_error("could not allocate");
```

Comment:
Improve error message to something the user might understand and know what do to about...